test/withdrawOptionRequest.ts

```diff
@@ -18,8 +18,22 @@ describe("NFTOpt Tests", function () {
     });
 
     describe("withdrawOptionRequest", function () {
-        it("tests that method can be called", async function () {
-            expect(NFTOptCTR.connect(buyer).withdrawOptionRequest(0)).to.not.throw;
+
+        it("should fail when option does not exist", async function () {
+            await expect(NFTOptCTR.connect(buyer)
+                .withdrawOptionRequest(9999))
+                .to.be.revertedWith("INVALID_OPTION_ID");
+        });
+
+        it("should fail when option is not in REQUEST state", async function () {
+            const option = await NFTOptCTR.options(1);
+            expect(option.state).to.equal(OptionState.Request);
```

Comment:
This test is not correct, it is passing only because `OptionState.Request` is equal to 0, which is the default value for the unexistent option retrieved above. To properly test this you have to actually publish a dummy option, which you should further create, so it will have another state.